stacks/ecs.yml

```diff
@@ -462,6 +468,112 @@ Resources:
         - !Ref VPCSubnet1
         - !Ref VPCSubnet2
         - !Ref VPCSubnet3
+  # ECS task draining
+  ECSDrainSNSTopic:
+    Type: "AWS::SNS::Topic"
+    DependsOn: ECSDrainFunction
+    Properties:
+      Subscription:
+        - Endpoint: !GetAtt ECSDrainFunction.Arn
+          Protocol: "lambda"
+  ECSDrainSNSRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              Service:
+                - "autoscaling.amazonaws.com"
+            Action:
+              - "sts:AssumeRole"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AutoScalingNotificationAccessRole
+      Path: "/"
+  ECSDrainLambdaRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      Policies:
+        - PolicyName: "drain-lambda-inline"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: "Allow"
+                Resource: "*"
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                  - ec2:DescribeInstances
+                  - ec2:DescribeInstanceAttribute
+                  - ec2:DescribeInstanceStatus
+                  - ec2:DescribeHosts
+                  - ecs:DescribeContainerInstances
+                  - ecs:DescribeTasks
+                  - ecs:ListContainerInstances
+                  - ecs:ListTasks
+                  - sns:ListSubscriptions
+              - Effect: "Allow"
+                Resource: !Ref ECSClusterASG
+                Action:
+                  - autoscaling:CompleteLifecycleAction
+              - Effect: "Allow"
+                Resource: !Ref ECSCluster
```

Comment:
May be the case that some of these , like the task state change, need access to subresources of the ECSCluster ARN. Couldn't find anythign obvious in the docs, so…we'll find out?